### PR TITLE
Change class name from CoordSysCartesian to CoordSys3D.

### PR DESCRIPTION
**Pull request by szymag**
Vector module is preparing to handle different type of curvilinear coordinate system, so main class `CoordSysCartesian` has misleading name. 
This PR changes only strings from `CoordSysCartesian` to `CoordSys3D`.

TODO:
- [x] update documentation

### CODE REVIEW

**Comment by szymag** [comment]
pin @Upabjojr

File: sympy/vector/tests/test_vector.py [hunk 1]
```diff
@@ -3,9 +3,9 @@
      sin, cos, Function, Integral, Derivative, diff, integrate
 from sympy.vector.vector import Vector, BaseVector, VectorAdd, \
      VectorMul, VectorZero
-from sympy.vector.coordsysrect import CoordSysCartesian
+from sympy.vector.coordsysrect import CoordSys3D
 
-C = CoordSysCartesian('C')
+C = CoordSys3D('C')
```
Comment (moorepants) [inline sympy/vector/tests/test_vector.py]: Tests for deprecated functions should be left in place until the deprecated function is removed. This ensures that the deprecated function doesn't break on any future changes to SymPy that occur as long as the deprecated function is in place.

Comment (Upabjojr) [inline sympy/vector/tests/test_vector.py]: I've got the impression that deprecation warnings are treated as errors in the tests.

Comment (moorepants) [inline sympy/vector/tests/test_vector.py]: You have to filter them so that the travis tests don't error but simply emit the warning.

Comment (moorepants) [inline sympy/vector/tests/test_vector.py]: There are a number of examples. Use this grep:
```
grep -rn "category=SymPyDeprecationWarning" sympy --include \*.py
```

Comment (Upabjojr) [inline sympy/vector/tests/test_vector.py]: @szymag could you add a test showing that the deprecated version behaves exactly like CoordSys3D? You should use filter warning.

Comment (szymag) [inline sympy/vector/tests/test_vector.py]: OK, I will check how to use filter checking.

Comment (Upabjojr) [inline sympy/vector/tests/test_vector.py]: I think we just need one test to check `assert CoordSysCartesian("C") == CoordSys3D("C")`, maybe also repeated with some optional arguments.

Comment (moorepants) [inline sympy/vector/tests/test_vector.py]: Yes, one is sufficient here.

Comment (szymag) [inline sympy/vector/tests/test_vector.py]: > I think we just need one test to check assert CoordSysCartesian("C") == CoordSys3D("C"), maybe also repeated with some optional arguments.

I have the same idea

**Pull request merged**